### ISSUE TITLE
Rule `no-setup-props-reactivity-loss` in Vue 3.5 should no longer trigger a warning when destructuring defineProps.

### ISSUE THREAD

**Issue by Heggria** [vuejs/eslint-plugin-vue#2543]
**What rule do you want to change?**

`no-setup-props-reactivity-loss`

**Does this change cause the rule to produce more or fewer warnings?**

fewer

**How will the change be implemented? (New option, new default behavior, etc.)?**

In [vue 3.5](https://blog.vuejs.org/posts/vue-3-5), variables destructured from props will retain reactivity, so this warning is incorrect in this context.

**Please provide some example code that this change will affect:**

[Playground](https://ota-meshi.github.io/eslint-plugin-vue-demo/#eJyNWEty5DYMvYqqN5NUmeVKKquJM8kRcgBvKAmSOKZIDUl1u+P47nmgvtTH8aolAAJB4OHXb5fClnT5enkK1HZaBvr2bLLsqVTX7BGPT48r+pMvnOpC5in0Xaalqf94vgT/fAHv2RTW+JC9ZUa29JA1pOomZH9kX3757ctDdlNlaMa37B0PJVXK0N/Odv7pjY/M4odfMx+cMvXvA2nQ8udEzf7NTN/m5EZ21HrIff/2089s/WAx7Ls8XFyvyV++vl2uPT0Wtm3JBFEqR0VQV/jg14fI+e5fRe/Ji6t0fqa2vQ5K3KwrBT7trOGP2eJFxFghnbM3UfUGKuEOoYy4yVA0iYy/m4I5rKcPVIoOXiAXVKqraJTmw0zAUQlj/x2rK2WQa7GSOlxNsiCzhM2/46YgF1o6yfadSRP8ae940qqi4l5oOpO0Grog5mEjOS9kpz4W9QVMLr224SNhuuLKHwlUSgdyZ8zR/VKLCbwnkk1otSBNEQnKn0gpowHV/9PFwfAgVBJIEaE5VecsgsdOMy8iyDp+eCIavSVkALzz/vRk9ubnhD6n7ypyZfAFYHoqcXq7q7AGecEZJVpbqkqdxmkQjRk7i56q7YlToVK1eKE716xUsse9wNgTr4ivJ6GqDUcrVrz4IvmSXjvrAifVUD9ErHhbCaiVFYdS3qRKMnTOmzOBtg9wkRmCnxzdoegwg1BJEsc5qlA6BGe8NGXK8eSu9GFhmmW2TpoZO0t8I12idFtgvCpJUIWrBv+JejblD5uQ0l8DTpJwig+yaBJY9galeGWEP2AmdZrJnna23BGTFyCSQmPLrTDgXImbCg2eqtTrYKNp+MX4Y8DE75IbMrZjBU/F+LL4iYXHmuVea6nYMQ6Rg3SoQFRthGflbMuJMbMd/ejRzlYhW6XpxGSHiHDvYj6h0PdFWF15knJkSpzkgHtndtyxfvul0+1Egq1r4F+ZiJIAzHqV9JxJcHDdGhMzR2pVzhU1KZODXUegux/IUKuCH9TJ9WWH2CMM9CqLxcvDuXOgYzJtmMP0Mig+ZqXpNLAMBz+o4mVDn3HjrN2aMRTjHbHQVm7V7Evdmr4jrpE+0RiXO+KBQrW92hWp1do9EYV6r3CFgoVU7C0E3Hc039jbnqgP/MapFokrnAuPnjBcm1E3Dn+jPTWFkGJ+ntYQzVbBt/RKRT9BiEcxCeJqovIzQ0vPGeJVPatiqKLQAQa1+IecXejotltRmHxEk0hqAEK6O7cgFG8/Wb+WYI1Ltg0cjHQad6vpdSKWlPd1TW551xQ47RYKd1Dpap8Qhrsh3vnUryfOGn+7rpz0XXbcRAX6x7SdXlA54dCCoxEPS5md5DiZlOi5oRYi13ZMr9ikNy4EIpwUubWapGEbwsSppNahwWhWNzMJtW3zfa1tjqkyJY7VeEM0Awyjw7uZ6vDe8xx8axRaSCeL2Qvawqm2YtAXah1VAA/mljwUnLjF0G0auYw127LOAtbYIsByQFW1+CVfyG4+GpsBnKH1rDAKTy/AfrCxYeQ9xm61gBzjQYQ+LQTcVsRrzUJooNXGOf4o0TBulNidHC8fquCKvgwxzObBiMHo5IInnrJFTkhlWmV2nAxQhpcXhICiyrjF8TS/8DB5wK25XpG8rAgrBhYIfd9QkbtynVgj2XbjwgFVyiBUCz+OKFrmtLh3JHZOXWMuHOXTdrZZzSK5LF7iIEBmgc/ELOZ1c0VMw82TTnyeGu1d0TincEdU3sixQA/dCaG3SwtY+mJz7xoy6Sa5mha5D8YiFAPJ2Tb5hQUr5fxqa+F1iHLEYmlqcUZCn4vjcA7cv1BgqM/hOxViAK7PGtY8FFmsWz6loreup69I+9Hb9XgXaRHF4zkzp5Wvq3lQdMNWtxg3oy1dNMeVfneXaf6FUUiPzuro2t1t4vrAf0as8myayCMFSWKRAALRBSI58eJ4frj2zaPHkH4zCzFcjZF8M6zcy5dxiDwK6wSpk6ltYCId0QZU2MxPuwl1YfEZmj7vymmFDXe9JvKox38tHEI3srdfDGtzQlxF3Loy3W61PfmjIEas0zQu667erA/J4BXVTsPtZu2JJQ+so0Piljr+EcFlDR1B8eZjY6t4f7jEGgqDL39F9w67LXm4LjyOvPf/AA2HBBo=)

```vue
// before: warning "Destructuring the `props` will cause the value to lose reactivity."
const { name, className = '', height = '14', width = '14' } = defineProps<{
    name: string
    className?: string
    height?: string | number
    width?: string | number
}>()

// after: no warning in vue 3.5 
const { name, className = '', height = '14', width = '14' } = defineProps<{
    name: string
    className?: string
    height?: string | number
    width?: string | number
}>()
```

**Comment by n0099** [comment]
https://github.com/vuejs/eslint-plugin-vue/issues/2121

**Comment by FloEdelmann** [comment]
Thanks for linking that issue! I'll close this as a duplicate then.
